packages/odf/modals/s3-browser/delete-and-empty-bucket/EmptyBucketModal.tsx

```diff
@@ -0,0 +1,308 @@
+import { useModal } from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'i18next';
+import { Trans, useTranslation } from 'react-i18next';
+import {
+  Modal,
+  Button,
+  Text,
+  TextInput,
+  ModalVariant,
+  ValidatedOptions,
+  TextInputTypes,
+  TextVariants,
+  TextContent,
+  Alert,
+  AlertVariant,
+  AlertActionCloseButton,
+  AlertActionLink,
+  FormGroup,
+} from '@patternfly/react-core';
+import { LazyDeleteObjectsSummary } from '../delete-objects/LazyDeleteModals';
+import { LazyDeleteBucketModal } from './lazy-delete-and-empty-bucket';
+
+type EmptyBucketModalProps = {
+  bucketName: string;
+  noobaaS3: S3Commands;
+  refreshTokens?: () => void;
+  setEmptyBucketResponse: React.Dispatch<
+    React.SetStateAction<EmptyBucketResponse>
+  >;
+};
+
+export const getTextInputLabel = (t: TFunction, bucketName: string) => (
+  <Trans t={t}>
+    <b>
+      To confirm deletion, type <i>{{ bucketName }}</i>:
+    </b>
+  </Trans>
+);
+
+const EmptyBucketModal: React.FC<CommonModalProps<EmptyBucketModalProps>> = ({
+  closeModal,
+  isOpen,
+  extraProps: { bucketName, noobaaS3, refreshTokens, setEmptyBucketResponse },
+}) => {
+  const { t } = useCustomTranslation();
+  const [inputValue, setInputValue] = React.useState('');
+  const [inProgress, setInProgress] = React.useState(false);
+
+  const onEmpty = async (event) => {
+    event.preventDefault();
+    setInProgress(true);
+    let deleteResponse: DeleteObjectsCommandOutput;
+
+    try {
+      let isTruncated = true;
+      let keyMarker: string;
+      let versionIdMarker: string;
+
+      while (isTruncated) {
+        const deleteObjectKeys = [];
+        const searchParams = {
+          Bucket: bucketName,
+          KeyMarker: keyMarker,
+          VersionIdMarker: versionIdMarker,
+        };
+
+        // eslint-disable-next-line no-await-in-loop
+        const objects = await noobaaS3.listObjectVersions(searchParams);
+
+        if (objects?.Versions) {
+          deleteObjectKeys.push(
+            ...objects.Versions.map((object) => ({
+              Key: object.Key,
+              VersionId: object.VersionId,
+            }))
+          );
+        }
+        if (objects?.DeleteMarkers) {
+          deleteObjectKeys.push(
+            ...objects.DeleteMarkers.map((marker) => ({
+              Key: marker.Key,
+              VersionId: marker.VersionId,
+            }))
+          );
+        }
+
+        // eslint-disable-next-line no-await-in-loop
+        deleteResponse = await noobaaS3.deleteObjects({
+          Bucket: bucketName,
+          Delete: { Objects: deleteObjectKeys },
+        });
+
+        if (deleteResponse.Errors?.length > 0) {
+          throw deleteResponse.Errors;
+        }
+
+        isTruncated = objects.IsTruncated;
+        keyMarker = objects.NextKeyMarker;
+        versionIdMarker = objects.NextVersionIdMarker;
+      }
+    } catch (err) {
+      console.error('Error while deleting bucket ', err);
```

Comment:
also I don't think we should log all errored objects to the console...
```suggestion
      console.error('Error while deleting bucket ');
```